packages/python/plotly/plotly/express/_core.py

```diff
@@ -1029,6 +1020,16 @@ def _escape_col_name(df_input, col_name, extra):
     return col_name
 
 
+def to_unindexed_series(x):
+    """
+    assuming x is list-like or even an existing pd.Series, return a new pd.Series with
+    no index, without extracting the data from an existing Series via numpy, which
+    seems to mangle datetime columns. Stripping the index from existing pd.Series is
+    required to get things to match up right in the new DataFrame we're building
+    """
+    return pd.Series(x).reset_index(drop=True)
```

Comment:
this makes a copy of the series because of the `reset_index` step, so it's suboptimal for non datetime columns. Could a solution be to return `x.values` in other cases?

Reply:
Hmmm yeah we weren't copying before but we are now. After spending like 8 hours trying various permutations here I think I'm OK with the copy on the principle of "make it correct, then make it fast". We can optimize it later if this really causes problems.